horace_core/sqw/@Experiment/Experiment.m

```diff
@@ -175,146 +90,256 @@
                 end
             else
                 error('HORACE:Experiment:invalid_argument', ...
-                    'Must give all of detector_array, instrument and sample')
+                    'Must give all of detector_array, instrument and sample or the structure representing them')
             end
         end
-        
-        function oldhdrs = convert_to_old_headers(obj)
-            nruns = numel(obj.expdata);
-            oldhdrs = cell(nruns,1);
-            edflds = fields(obj.expdata);
-            for i=1:nruns
-                oldhdr = struct();
-                for j=1:numel(edflds)
-                    oldhdr.(edflds{j}) = obj.expdata(i).(edflds{j});
-                end
-                oldhdr.alatt = obj.samples(i).alatt;
-                oldhdr.angdeg = obj.samples(i).angdeg;
-                if isa(obj.instruments(i),'IX_null_inst')
-                    oldhdr.instrument = struct();
-                else
-                    oldhdr.instrument = obj.instruments(i);
-                end
-                if isa(obj.samples(i),'IX_null_sample')
-                    oldhdr.sample = struct();
-                    oldhdr.alatt = obj.samples(i).alatt;
-                    oldhdr.angdeg = obj.samples(i).angdeg;
-                else
-                    oldhdr.sample = obj.samples(i);
+        %
+        function oldhdrs = convert_to_old_headers(obj,header_num)
+            % convert Experiment into the old header structure, initially
+            % stored within binary files
+            samp = obj.get_unique_samples();
+            if nargin == 2
+                oldhdrs = obj.expdata_(header_num).to_bare_struct();
+                oldhdrs.alatt = samp.alatt;
+                oldhdrs.angdeg = samp.angdeg;
+                oldhdrs.instrument = struct();
+                oldhdrs.sample = struct();
+            else
+                nruns = obj.n_runs;
+                oldhdrs = cell(nruns,1);
+                for i=1:nruns
+                    old_hdr = obj.expdata_(i).to_bare_struct();
+                    old_hdr.alatt = samp.alatt;
+                    old_hdr.angdeg = samp.angdeg;
+                    
+                    old_hdr.instrument = struct();
+                    old_hdr.sample = struct();
```

Comment:
that what was happening in old binary files. The headers contain rubbish on the instrument place and real instrument and sample are stored separately at the end of the files. The conversion occurs at save/load operation. As the useful purpose of `to_old_headers`  method is provide compatibility with old binary formats, we should do just that. 

If there are other usages of the method, this may change, but should it? All conversation to the word should happen through `Experiment` interface